packages/reactivity/src/dep.ts

```diff
@@ -387,5 +387,7 @@ export function getDepFromReactive(
   key: string | number | symbol,
 ): Dep | undefined {
   const depMap = targetMap.get(object)
+  // #12427, while the target object is an array, the dep key(array index) must be string
```

Comment:
I would fix it here https://github.com/vuejs/core/blob/14f6917c3c47b9e27298c741e0e3f76c89868482/packages/reactivity/src/baseHandlers.ts#L113

```ts
- track(target, TrackOpTypes.GET, key)
+ track(target, TrackOpTypes.GET, targetIsArray && isIntegerKey(key) ? +key : key)
```

Reply:
thanks, I will do like this.

Reply:
I tried to do it this way, but it caused a series of other problems. Many unit tests will not pass, which may be a breaking change.

Reply:
ok, thanks for checking. pls go ahead with the PR